### PR TITLE
Migrating to Java11 and Gradle 7.4

### LINKED ISSUE #741: Update to Java 11 as minimum required version
Version 6.0 of HAPI (in prerelease as of the time of this writing) requires Java 11. The cql-translator uses some HAPI APIs to generate data-requirements for CQL files. In order to update to the latest HAPI releases, the cql-translator will need to be updated to 11 as a minimum as well.

https://hapifhir.io/hapi-fhir/docs/introduction/changelog.html#hapi-fhir-600-prerelease

To do the upgrade we need to change gradle build to the correct Java version, ensure the CI pipeline is updated to use the correct version, and gather feedback from the community on the timing of the change from 8 to 11.

### PR DESCRIPTION
**Pull request by vitorpamplona**
Fixes #741: Updates the source code compatibility requirements to Java 11. 

Since Java11 removes JAXB, the upgrade requires migrating `JAXB commons` from `0.12.0` to `1.11.1` and the JAXB runtime dependency below to the `xds-to-modelInfo` module: 
- ```    implementation group: 'org.jvnet.jaxb2_commons', name: 'jaxb2-basics-runtime', version: '1.11.1'```

Related to #582 and #512, I also tested migrating to Gradle from `5.6.2` to `7.4.2`, which requires us to: 
1. Rework the [deprecated maven publishing configuration](https://docs.gradle.org/6.9.2/userguide/upgrading_version_5.html#legacy_publication_system_is_deprecated_and_replaced_with_the_publish_plugins) to use the [new plugin](https://docs.gradle.org/current/userguide/publishing_maven.html)
2. Update a these scopes: 
  - `compile` to `implementation`
  - `testCompile` to `testImplementation`
  -  `runtime` to `runtimeOnly`
  
This PR also updates the lib to use HAPI 6.0.1 and FHIR core 5.6.36. 
  
I wasn't able to successfully test the new Maven Publishing plugin. Maybe we do a few snapshot releases to make sure everything is being signed and is correctly packaged?

### CODE REVIEW

**Comment by brynrhodes** [comment]
@vitorpamplona , looks like the most recent merges resulted in some conflicts, can you resolve those when you get a chance?

**Comment by vitorpamplona** [comment]
Ready to go, @brynrhodes! You might have some issues with the publishing plugin. I did my best to translate to the new version, but couldn't really test it.

File: Src/java/build.gradle [hunk 1]
```diff
@@ -183,79 +185,89 @@ configure(subprojects.findAll {it.name in ['model', 'elm', 'quick', 'qdm', 'cql-
...
+                    password project.hasProperty("ossrhPassword") ? ossrhPassword : ""
+                }
+
+                /* Use these to test locally (but don't forget to comment out others!)
+                def releasesRepoUrl = "file://${buildDir}/repo"
+                def snapshotsRepoUrl = "file://${buildDir}/ssRepo"
+                */
+
+                // change URLs to point to your repos, e.g. http://my.org/repo
+                def releasesRepoUrl = "file://${buildDir}/repo" //"https://oss.sonatype.org/service/local/staging/deploy/maven2/"
```
Comment (brynrhodes) [inline Src/java/build.gradle:250]: This works for me locally if I just use the commented out URLs, so go ahead and uncomment them and commit that to the PR and I'll merge it

**Review by brynrhodes — CHANGES_REQUESTED** [review]

**Review by JPercival — COMMENTED** [review]
HAPI is migrating to Java 11 being required as of 6.0. Do we want to make this change at the same time for the sake of being consistent across the different frameworks? Also, this is a backwards incompatible change. Do we need to consider updating project major versions? It's currently tied to the cql specification level (i.e. 1.5).

**Comment by brynrhodes** [comment]
So @JPercival , are you saying we should also update to HAPI 6.0 with this change?

**Comment by JPercival** [comment]
Other way around. I'm saying we should do this when we update to 6.0. None of the rest of the stack is stable right now due to fhir core and hapi mismatches that got propagated through. I don't want to add a 6.0 change on top of that at the moment, myself.

**Comment by brynrhodes** [comment]
I see, so keep this PR on the backlog for now, agreed. Vitor, can you mark this PR draft until we switch to HAPI 6.0?

**Comment by JPercival** [comment]
@vitorpamplona - What's the state of Java 11 support on Android?

**Comment by vitorpamplona** [comment]
We already use Java11 on the SDK. Google requires Java11 for Android Dev since July 2021. It's very stable at the moment.

**Comment by JPercival** [comment]
Can I change my mind and suggest we go ahead and roll this in since we decided to do a major release on the other PR? There are at least 3 breaking changes under consideration:

* Java 11 - New platform requirements
* Jackson Serialization - Breaks old JSON ELM
* Complier/Translator split - Public API changed

Might as well get them all done. It'll take me a while to update downstream projects but the translator has wider usage so it's probably best to minimize churn here.

**Comment by JPercival** [comment]
@vitorpamplona - Can you update this PR to HAPI 6.0.1 and FHIR core 5.6.36?

https://github.com/hapifhir/hapi-fhir/blob/v6.0.1/pom.xml#L757

**Comment by JPercival** [comment]
@mdnazmulkarim - Can you take a look at this one too?

**Comment by vitorpamplona** [comment]
@JPercival Migration to HAPI 6.0.1 and FHIR core 5.6.36 done.

**Review by mdnazmulkarim — APPROVED** [review]

**Comment by JPercival** [comment]
@vitorpamplona - I see some conflicts. Can you resolve and I'll merge?

**Pull request merged**